sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -341,6 +347,33 @@ class FileStreamSourceSuite extends FileStreamSourceTest with SharedSQLContext {
     Utils.deleteRecursively(tmp)
   }
 
+  test("metadataPath should be in checkpointLocation") {
```

Comment:
Could we just test this in DataFrameReaderWriterSuite?  This seems kind of integration heavy.  It would be good to test that multiple sources get different ids too.